### PR TITLE
Add Espressif ESP32-S2-DevKitC-1-N8R2 variant

### PR DESCRIPTION
**Pull request by kylefmohr**
This is a relatively new board, [purchased from here](https://www.amazon.com/dp/B0B97K3LJQ), apparently first released August 10th 2022. The official pinout is [available here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/_images/esp32-s2-devkitc-1-v1-pinout.png). It is very similar to the N4R2 version that is available for CircuitPython.

I got the majority of my code from #5998, so thanks to anecdata for that!

I would appreciate if somebody could review my changes to ci_check_duplicate_usb_vid_pid.py, as this is my first PR for this project, and I'm a little unclear on how that file works. With a micro USB cable plugged into the "UART" port of the board, this is what shows up in Device Manager in Windows 
image

I think I added the PID/VID correctly, just want another set of eyes on them. Thanks!

Edit: ~~Marking as draft until build completes.~~

Edit2: Builds don't happen if it's just a draft 🤦‍♂️

### CODE REVIEW

File: ports/espressif/boards/espressif_esp32s2_devkitc_1_n8r2/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
+USB_VID = 0x10C4
+USB_PID = 0xEA60
```
Comment (Neradoc) [inline ports/espressif/boards/espressif_esp32s2_devkitc_1_n8r2/mpconfigboard.mk:2]:
```suggestion
USB_VID = 0x303A
USB_PID = 0x7009
```

File: tools/ci_check_duplicate_usb_vid_pid.py [hunk 2]
```diff
@@ -56,6 +56,7 @@
         "espressif_esp32s3_devkitc_1_n8r8",
     ],
     "0x303A:0x7009": ["espressif_esp32s2_devkitc_1_n4", "espressif_esp32s2_devkitc_1_n4r2"],
+    "0x10C4:0xEA60": ["espressif_esp32s2_devkitc_1_n8r2"],
```
Comment (Neradoc) [inline tools/ci_check_duplicate_usb_vid_pid.py:59]:
```suggestion
    "0x303A:0x7009": ["espressif_esp32s2_devkitc_1_n4", "espressif_esp32s2_devkitc_1_n4r2", "espressif_esp32s2_devkitc_1_n8r2"],
```

**Review by Neradoc — CHANGES_REQUESTED** [review]
What you are showing is the VID/PID of the CP2104 chip on the UART port. The VID and PID set in Circuitpython is for the native USB port of the board, and must be unique (mostly...) and properly allocated by a VID owner.
In this case, it seems that you should use the common `espressif_esp32s2_devkitc_1` PID allocated by Espressif, and therefore add it to the existing list for that VID/PID in the duplicate check.
I made the changes in the suggestions.
I didn't look at the other files.

File: tools/ci_check_duplicate_usb_vid_pid.py (hunk 2 above)
Comment (kylefmohr) [reply tools/ci_check_duplicate_usb_vid_pid.py:59]: Thanks, this is exactly the feedback I was looking for! 

When I use the native USB port on the board instead, I show a VID of 303A as expected, but a PID of 0002, should I create a new entry for that PID?

Comment (Neradoc) [reply tools/ci_check_duplicate_usb_vid_pid.py]: no, Circuitpython defines the pid and vid, what you see in windows has nothing to do with that. Use the changes I indicated.

Comment (kylefmohr) [reply tools/ci_check_duplicate_usb_vid_pid.py]: Will do, thanks!

**Comment by dhalbert** [comment]
@kylefmohr Thanks for this!

I pushed a minor change to fix the formatting of `tools/ci_check_duplicate_usb_vid_pid.py`.

It's a good idea to create a separate branch for a PR instead of using your `main`, because now your `main` has diverged from upstream. You may want to force it back to `adafruit/main` later.

**Comment by kylefmohr** [comment]
> @kylefmohr Thanks for this!
> 
> I pushed a minor change to fix the formatting of `tools/ci_check_duplicate_usb_vid_pid.py`.
> 
> It's a good idea to create a separate branch for a PR instead of using your `main`, because now your `main` has diverged from upstream. You may want to force it back to `adafruit/main` later.

Thank you! That makes sense. In the past I've deleted and re-forked the repo, but creating a new branch for the PR seems easier. I appreciate the assistance.

**Comment by dhalbert** [comment]
We have a Learn Guide explaining about branches: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github/always-work-on-a-branch (the whole guide is worth reading if you have not seen it yet).

**Comment by kylefmohr** [comment]
> We have a Learn Guide explaining about branches: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github/always-work-on-a-branch (the whole guide is worth reading if you have not seen it yet).

I have not, I'll definitely give that a read! Another newbie question that I don't see covered in the guide: How did you manage to **only** build the firmware for the new board, and skip all other boards/ports? That seems very useful, in the past I've just waited for the build to complete entirely.

**Comment by dhalbert** [comment]
Our scripting for the automated builds checks to see where the changed files are, and if they are only in a particular port or board, it will rebuild only those ports or boards. This was done in PR #5312.

**Comment by dhalbert** [comment]
@Neradoc Does this look good to you?

**Comment by Neradoc** [comment]
There's a define for AUTORESET_DELAY_MS in mpconfigboard.h that is unnecessary. The rest looks good to go.
(To quote  #6182: _it never did anything but was introduced somehow_)

**Review by Neradoc — APPROVED** [review]

**Comment by dhalbert** [comment]
I will fix that, and will also shorten `board.c`, due to the new `MP_WEAK` default versions of those routines.

**Review by Neradoc — APPROVED** [review]

**Comment by Neradoc** [comment]
Ah we are crossing streams !

**Review by Neradoc — APPROVED** [review]

**Pull request merged**